core/domain/skill_services.py

```diff
@@ -1377,13 +1413,19 @@ def get_categorized_skill_ids_and_descriptions(
     for topic in topics:
         subtopics = topic.subtopics
         for skill_id in topic.uncategorized_skill_ids:
-            categorized_skills.add_uncategorized_skill(  # type: ignore[no-untyped-call]
+            description = skill_descriptions[skill_id]
+            # Ruling out the possibility of None for mypy type checking.
+            assert description is not None
```

Comment:
Ditto, needs justification.

Reply:
above we are using `get_descriptions_of_skills()` method and this method returns a tuple. In this tuple we have 2 items, first one is a dictionary -- in this dictionary, we have a skill id as a key and value as a corresponding skill_id's skill description but if a skill corresponding to a particular skill_id is deleted then its description value is None. And the second item in the tuple is a list of all deleted skill_ids. 

So here, at line 1422 description can only be None when its corresponding skill does not exist and the main motive of this function is to Categorize skills so if a skill does not exist then there is no point to Categorize a non-existing skill.  

My solution -- We should skip those cases where skill does not exist, I mean skip those cases where the description value is none.

Reply:
I'm a little confused. Shouldn't the description always be a string (possibly the empty string) -- isn't that what happens on creation? It seems odd to change the description to None solely because of a deletion -- I would expect it to be still typed as str throughout because I don't think I see what None represents exactly (that couldn't be represented by, say, the `deleted=True` field).

Reply:
If your mean is that we should change the non-existing skill's description from None to '' (empty string) then I'm not sure about this change because the `get_descriptions_of_skills` function is directly used inside the controller and there we are not handling the case of none and passing the value to front-end directly ( if I'm not wrong ).

https://github.com/oppia/oppia/blob/4ba04d87b7e8aea3953287ae183c8e7d1c0031b6/core/controllers/topic_editor.py#L347
https://github.com/oppia/oppia/blob/4ba04d87b7e8aea3953287ae183c8e7d1c0031b6/core/controllers/topic_editor.py#L438
https://github.com/oppia/oppia/blob/4ba04d87b7e8aea3953287ae183c8e7d1c0031b6/core/controllers/topic_viewer.py#L104

Reply:
In frontend, only Dict[str, str] is expected,
https://github.com/oppia/oppia/blob/b1ec86cb2b3471530e74840dabb5755ea1b2a692/core/templates/domain/topic/subtopic.model.ts#L33

So, as per the discussion in chat I changed the implementation of `get_descriptions_of_skills` a bit. Now, assert is removed. PTAL!

Reply:
It seems to me like both the assert and the comment are still there.

Reply:
forgot to remove them , now Done!